MRG: Refactor plot_alignment

GuillaumeFavelier opened the pull request:
This PR refactors `plot_alignment()` following https://github.com/mne-tools/mne-python/pull/9689#issuecomment-911385945. It  fixes https://github.com/mne-tools/mne-python/issues/9542. It fixes #9715.

It's still work in progress.

**ToDO**
- [x]  _plot_head_surface
- [x] _plot_axes
- [x] _plot_head_shape_points
- [x] _plot_head_fiducials
- [x] _plot_mri_fiducials
- [x] _plot_forward

The functions should all receive an instance of `_Renderer` as first parameter and return a list of `vtkActor`s as first output.

larsoner commented:
If possible could you also fix #9715 / https://github.com/mne-tools/mne-python/pull/9585#issuecomment-910546309 ?

GuillaumeFavelier commented:
I updated the branch with `main` and I think I fixed it indirectly.

https://33649-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/forward/35_eeg_no_mri.html?highlight=eeg%20mri

agramfort reviewed (commented):
so far so good !

@GuillaumeFavelier tell me when it's good enough from your end.

agramfort commented on mne/viz/_3d.py:
```diff
@@ -986,9 +966,36 @@ def _plot_head_shape_points(renderer, info, to_cf_t):
+                          fwd_nn[:, ori, 0],
+                          fwd_nn[:, ori, 1],
+                          fwd_nn[:, ori, 2],
+                          color=color, mode='arrow', scale=1.5e-3)
```
you should return something no?

GuillaumeFavelier replied:
Done in fa90c93 !

GuillaumeFavelier commented:
It's now ready on my end @agramfort, what do you think?

larsoner commented:
Also @alexrockhill feel free to look if you have time

alexrockhill commented:
Didn't run it but looks like great fixes, thanks and sorry I didn't have the time to do the bug I made
